README.md

```diff
@@ -1,2 +1,283 @@
+<p align="center">
+  <a href="https://fingerprintjs.com">
+    <img src="resources/logo.svg" alt="FingerprintJS" width="300px" />
+  </a>
+</p>
+
+[![](https://jitpack.io/v/fingerprintjs/fingerprint-android.svg)](https://jitpack.io/#fingerprintjs/fingerprint-android)&nbsp;&nbsp;![Test](https://github.com/fingerprintjs/fingerprint-android/workflows/Test/badge.svg)&nbsp;&nbsp;[![API](https://img.shields.io/badge/API-19%2B-brightgreen.svg?style=plastic)](https://android-arsenal.com/api?level=19)
+# fingerprint android
+
+Lightweight library for device identification and fingerprinting.
+
+Fully written in Kotlin. 100% Crash-free. 
+
+Creates a user identifier from all available platform signals.
+
+The identifier is fully stateless and will remain the same after reinstalling or clearing application data.
+
+# Table of Contents
+1. [Quick start](#quick-start)
+2. [API Reference](#api-reference)
+3. [Initialization and usage](#initialization-and-usage)
+4. [Advanced usage](#advanced-usage)
+
+
+## Quick start
+
+### Install as Gradle dependency
+
+Add these lines to `build.gradle` file of your project.
+
+
+```gradle
+allprojects {
+	repositories {
+		...
+		maven { url 'https://jitpack.io' }	
+	}
+}
+```
+
+### Add the dependency to your project
+
+Add these lines to `build.gradle` file of your module.
+This library depends on `kotlin-stdlib`. If your application is written in Java, add `kotlin-stdlib` dependency first (it's lightweight and has excellent backward and forward compatibility).
+
+```gradle
+dependencies {
+	implementation 'com.github.fingerprintjs:fingerprint-android:1.0.0'
+	
+	// Add this line for java-project
+	implementation "org.jetbrains.kotlin:kotlin-stdlib:$kotlin_version"	
+}
+```
+
+## API Reference
+
+### Device ID and Fingerprint
+
+The library operates with two entities. 
+
+1. `deviceId` - it is an ID of the device, based on system-provided IDs. Currently, it is based on Google Service Framework ID, and ANDROID_ID if the first is unavailable. It's strict, which means the uniqueness of the ID for every device. Also, it keeps the same after application reinstalling.
+2. `fingerprint` - it is a digital fingerprint of a device. Basically, it is a hash, calculated from a lot of available platform signals. `fingerprint` **is not a strict ID**, so there is a probability that two different devices will have the same fingerprint. Also, there is a probability, that the same device will have a different fingerprint in different moments due to system update, settings changing, etc.
+
+#### Why do we need them both?
+It is a good question, most of use-cases can be covered with the `deviceId`. But the `GSF_ID` and `ANDROID_ID`can be spoofed much easier, than device data such as CPU info or sensors list. 
+
+Another reason for using `fingerprint` is a possible restriction of access to `deviceId` due to privacy policy changes trend.
```

Comment:
`deviceId` is guaranteed to be random and should be your first choice for device identification. This identifier can be spoofed though and shouldn't be used in security/fraud scenarios. 

`fingerprint` is much harder to spoof and is a safer choice in security-focused use cases.

Reply:
I would remove the line about privacy trends, we can add that when `deviceId` is actually restricted, but we shouln't have it now because it's still possible to use it.